code/modules/mob/living/simple_animal/abnormality/aleph/army_in_black.dm

```diff
@@ -0,0 +1,498 @@
+
+/mob/living/simple_animal/hostile/abnormality/army/Move()
+	return FALSE
+
+/***Work procs***/
+//protect work grants you a buff in exchange for reducing its counter
+/mob/living/simple_animal/hostile/abnormality/army/AttemptWork(mob/living/carbon/human/user, work_type)
+	..()
+	if(work_type == "Protection")
+		if(datum_reference?.qliphoth_meter > 1)
+			if(user in protected_targets)
+				return FALSE
+			datum_reference.qliphoth_change(-1)
+			if((get_user_level(user)) < 4)
+				user.adjustSanityLoss(500)
+				return FALSE
+			protected_targets += user
+			user.apply_status_effect(STATUS_EFFECT_PROTECTION)
+			to_chat(user, "<span class='nicegreen'>You feel like you're in good company.</span>")
+			playsound(get_turf(user), 'sound/abnormalities/armyinblack/pink_heal.ogg', 50, 0, 2)
+			FriendlyBreach(user)
+		return FALSE
+	return TRUE
+
+//counter goes down when people die, repression/protection work is performed, or raised when an abnormality is suppressed.
+/mob/living/simple_animal/hostile/abnormality/army/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
+	if(work_type == ABNORMALITY_WORK_REPRESSION)
+		datum_reference.qliphoth_change(-1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/army/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	datum_reference.qliphoth_change(-1)
+	return
+
+//death counter from MoSB, it works exactly the same.
+/mob/living/simple_animal/hostile/abnormality/army/proc/OnMobDeath(datum/source, mob/living/died, gibbed)
+	SIGNAL_HANDLER
+	if(!IsContained())//Stops if it's breaching
+		return FALSE
+	if(died.z != z)//check for z-level congruency
+		return FALSE
+	if(istype(died, /mob/living/simple_animal/hostile/abnormality))//counter restored if another abnormality dies
+		datum_reference.qliphoth_change(1)
+		return TRUE
+	if(!died.mind)//if it's not a player
+		return FALSE
+	if(!ishuman(died))//all simple mobs, ordeals, other npcs except for humans
+		return FALSE
+	death_counter += 1
+	if(death_counter >= 2)
+		death_counter = 0
+		datum_reference.qliphoth_change(-1)
+	return TRUE
+
+/mob/living/simple_animal/hostile/abnormality/army/proc/FriendlyBreach(mob/living/carbon/human/user)
+	var/turf/T = get_turf (src)
+	var/mob/living/simple_animal/hostile/army_friend/A = new(get_turf(T))
+	friend_army += A
+	A.protection_target = user
+	A.GiveTarget(user)
+
+//*--Combat Mechanics--*
+
+/mob/living/simple_animal/hostile/abnormality/army/BreachEffect(mob/living/carbon/human/user)
+	..()
+	Blackify()
+	SpawnAdds()//set its alpha to 0 and make it non-dense
+	for(var/mob/living/L in protected_targets)
+		L.remove_status_effect(STATUS_EFFECT_PROTECTION)
+	density = FALSE
+	alpha = 0
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/mob/living/simple_animal/hostile/abnormality/army/proc/SpawnAdds()
+	var/list/number_of_adds = list()
+	var/list/spawns = shuffle(GLOB.xeno_spawn)
+	for(var/i = LAZYLEN(number_of_adds), i < adds_max, i++)//# of iterations is equal to adds_max
```

Comment:
This uh, this doesn't update in real time homedog. This is the STARTING value. Im pretty sure this entire list is redundant.